packages/alfa-rules/src/common/predicate/has-role-depending-on-name.ts

```diff
@@ -0,0 +1,16 @@
+import { Device } from "@siteimprove/alfa-device";
+import { Element } from "@siteimprove/alfa-dom";
+import { Predicate } from "@siteimprove/alfa-predicate";
+
+import { hasExplicitRole, hasAccessibleName } from "../predicate";
+
+const { hasName } = Element;
+const { and, not } = Predicate;
+
+export function hasRoleDependingOnName(device: Device): Predicate<Element> {
```

Comment:
```suggestion
// <form> and <section> element have an implicit role only if they have an accessible name
// Alfa currently can't handle that and always give them a role.
// see https://github.com/Siteimprove/alfa/issues/298
//
// This detects the cases where such a role was incorrectly given.
export function hasRoleDependingOnName(device: Device): Predicate<Element> {
```

Reply:
Maybe let's make the name even more explicit like `hasIncorrectRoleWithoutName` 🤔